bba/bba.go

```diff
@@ -3,92 +3,134 @@ package bba
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
+	"log"
 	"sync"
 	"sync/atomic"
 
-	"github.com/DE-labtory/cleisthenes/pb"
+	"github.com/DE-labtory/iLogger"
+
 	"github.com/golang/protobuf/ptypes"
 
+	"github.com/DE-labtory/cleisthenes/pb"
+
 	"github.com/DE-labtory/cleisthenes"
 )
 
 var ErrUndefinedRequestType = errors.New("unexpected request type")
 
-type Binary = bool
-
-const (
-	one  Binary = true
-	zero        = false
-)
-
 type request struct {
 	sender cleisthenes.Member
-	data   *pb.Message_Bba
+	data   cleisthenes.Request
+	round  uint64
 	err    chan error
 }
 
 type BBA struct {
 	*sync.RWMutex
+	cleisthenes.Tracer
 
+	owner cleisthenes.Member
 	// number of network nodes
 	n int
-
 	// number of byzantine nodes which can tolerate
 	f int
 
-	proposer cleisthenes.Member
-
 	stopFlag int32
 	// done is flag whether BBA is terminated or not
 	done bool
-
 	// epoch is current honeybadger epoch
 	epoch uint64
 	// round is value for current Binary Byzantine Agreement round
 	round       uint64
 	binValueSet *binarySet
-
 	// broadcastedBvalSet is set of bval value instance has sent
 	broadcastedBvalSet *binarySet
-
 	// est is estimated value of BBA instance, dec is decision value
-	est, dec Binary
+	est, dec *cleisthenes.BinaryState
```

Comment:
`est`'s initial value is **undefined**. and when `bba.HandleInput` est is set!

Reply:
Then i think bba has to have method which can modify `undefined` to `zero` because acs should set bba's est as zero when bba instance with 1 match `n-f`.

Reply:
we can utilize `HandleInput`